### DISCUSSION TITLE
Release Notes - 4.0.0

### DISCUSSION THREAD

**Discussion by dahall** [dahall/Vanara#452] [General]
- Nullable notation added to every project
- **Lots of breaking changes** from that nullability as I adjusted code based on new constraints
  - I'm still deciding on if I will try to backport fixes applied to the 4.0 branch to master or fold them together. Your feedback is appreciated.
- Use of global using statements
- Addition of .NET 7.0 and 8.0
- Finally fixing and releasing VssApi
- Fixed #392, #439, #441, #442, #443, #445, #448, #450

**Full Changelog**: https://github.com/dahall/Vanara/compare/v3.4.17...v4.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/dahall/Vanara/releases/tag/v4.0.0'>Release Notes - 4.0.0</a>.</em>

**Comment by tajbender** [comment]
Works 🥇
